### ISSUE TITLE
Using CSS classes instead of style props

### ISSUE THREAD

**Issue by tohjustin** [Hacker0x01/react-datepicker#952]
Currently the `z-index` attribute is hardcoded into the style `props` of `<Popper />` 

```JSX
// src/popper_component.jsx
<Popper
  style={{zIndex: 1}}
  modifiers={popperModifiers}
  placement={popperPlacement}
  className="popper">
  {popperComponent}
</Popper>
```
(see https://github.com/Hacker0x01/react-datepicker/blob/master/src/popper_component.jsx#L63)

To allow the users to overwrite this without having to use `!important`, I think it's a better idea to apply the `z-index` via a CSS class.

**Comment by tohjustin** [comment]
Will attempt to submit a PR over this weekend!

**Comment by tohjustin** [comment]
PR merged in #955
